Add more context on the behavior of the GPU function (#1800)
FluxML/Flux.jl: docs/src/gpu.md

d(cu(rand(10)))
```

> As a convenience, Flux provides the `gpu` function to convert models and data to the GPU if one is available. By default, it'll do nothing. So, you can safely call `gpu` on some data or model (as shown below), and the code will not error, regardless of whether the GPU is available or not. If the GPU loads successfully, `CUDA` will cause the data to move to the GPU from the CPU. As is shown below, this will change the type of something like a regualar array to a `CuArray`.

Reviewer: ```suggestion
As a convenience, Flux provides the `gpu` function to convert models and data to the GPU if one is available. By default, it'll do nothing. So, you can safely call `gpu` on some data or model (as shown below), and the code will not error, regardless of whether the GPU is available or not. If the GPU library (CUDA.jl) loads successfully, `gpu` will move data   from the CPU to the GPU. As is shown below, this will change the type of something like a regular array to a `CuArray`.
```

> As a convenience, Flux provides the `gpu` function to convert models and data to the GPU if one is available. By default, it'll do nothing, meaning that you can safely call `gpu` on some data or model (as shown below), and the code will not error, regardless of whether the GPU is available or not. If the GPU loads successfully, `CUDA` will cause the data to move to the GPU from the CPU. As is shown below, this will change the type of something like a regualar array to a `CuArray`.

Reviewer 2: ```suggestion
As a convenience, Flux provides the `gpu` function to convert models and data to the GPU if one is available. By default, it'll do nothing. So, you can safely call `gpu` on some data or model (as shown below), and the code will not error, regardless of whether the GPU is available or not. If the GPU loads successfully, `CUDA` will cause the data to move to the GPU from the CPU. As is shown below, this will change the type of something like a regualar array to a `CuArray`.
```
